packages/cdr/docker/jsreport/scripts/report-4a/content.js

```diff
@@ -0,0 +1,254 @@
+const axios = require('axios')
+
+const ES_INDEX = 'fhir-report-patient'
+const ES_USERNAME = 'elastic'
+const ES_PASSWORD = 'dev_password_only'
+
+async function beforeRender(req) {
+  const { from, to, state, district, city, facilityCode } = req.data.params
+
+  const esQuery = {
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          {
+            range: {
+              'hivDiagnosis.hivPosDate': {
+                lte: `${to}||/d`
+              }
+            }
+          },
+          {
+            must_not: {
+              exists: {
+                field: 'death.date'
+              }
+            }
+          },
+          ...(state !== 'all' // only include this filter if not 'all'
+            ? [
+                {
+                  term: {
+                    'registration.facility.state': state
+                  }
+                }
+              ]
+            : []),
+          ...(district !== 'all' // only include this filter if not 'all'
+            ? [
+                {
+                  term: {
+                    'registration.facility.district': district
+                  }
+                }
+              ]
+            : []),
+          ...(city !== 'all' // only include this filter if not 'all'
+            ? [
+                {
+                  term: {
+                    'registration.facility.city': city
+                  }
+                }
+              ]
+            : []),
+          ...(facilityCode !== 'all' // only include this filter if not 'all'
+            ? [
+                {
+                  term: {
+                    'registration.facilityCode': facilityCode
+                  }
+                }
+              ]
+            : [])
+        ]
+      }
+    },
+    aggs: {
+      age: {
+        range: {
+          field: 'death.ageAtDeath',
```

Comment:
I'm getting no results for this report. I think it's because we are requiring `death.date` to not exist but are using the death age, from the report specification it state we need to use registration age. To get that age we need to work it out from birthDate.

Here is a link to how it has been done before using a script: https://github.com/jembi/disi-hiv-mvp/blob/master/packages/cdr/docker/jsreport/scripts/report-1a/content.js#L67